src/platform/datapath_raw.h

```diff
@@ -169,6 +170,32 @@ CxPlatSockPoolUninitialize(
     _Inout_ CXPLAT_SOCKET_POOL* Pool
     );
 
+//
+// Returns TRUE if the socket matches the given addresses. This code is used in
+// conjunction with the hash table lookup, which already compares local UDP port
+// so it assumes that matches already.
+//
+inline
+BOOL
+CxPlatSocketCompare(
+    _In_ CXPLAT_SOCKET* Socket,
+    _In_ const QUIC_ADDR* LocalAddress,
+    _In_ const QUIC_ADDR* RemoteAddress
+    )
+{
+    if (Socket->Wildcard) {
+        return TRUE; // The local port match is all that is needed.
```

Comment:
Let's be judicious about assuming the hashtable lookup is based on the local port, because I think that could change. In TCPIP.sys, a lot of pain has come out of making the portpool a local port based table rather than a 4tuple-based table.

Reply:
We can add a debug assert.

Reply:
Sounds good, but my comment is more about the code architecture- I'm saying keep this possible change in mind and avoid writing code that depends deeply on the local port being the HT key.

Reply:
I agree it's an assumption. While it may change in the future, I'm not so sure it will. So I'd rather go with the assert design that would immediately break if we change the assumption.

Reply:
Yeah, that sounds perfectly good to me.